src/main/java/net/snowflake/ingest/utils/HttpUtil.java

```diff
@@ -203,23 +206,38 @@ private static ServiceUnavailableRetryStrategy getServiceUnavailableRetryStrateg
       @Override
       public boolean retryRequest(
           final HttpResponse response, final int executionCount, final HttpContext context) {
-        int statusCode = response.getStatusLine().getStatusCode();
-        if (executionCount == MAX_RETRIES + 1) {
```

Comment:
Looks like MAX_RETRIES could be removed now?

Reply:
It is still used `HttpRequestRetryHandler`. It is used when no server response was produced (some connection timeouts, etc). There we don't have any control over retry interval, so I prefer to keep it as it is and not introduce this waiting for 2 minutes.